portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,13 +20,91 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+
+import java.util.*;
+
+/** Servlet that interacts with a Google DataStore database for a comments section */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
-}
+    /**
+     * Gets database data for comments
+     * @param request The request object 
+     * @param response The response object
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Gets possible limit to the maximum number of comments  
+        String commentLimitString = request.getParameter("limit");
+        Integer commentLimitInteger = tryParseInt(commentLimitString);
```

Comment:
I think you can do int commentLimitInteger = tryParseInt(commentLimitString); Since if the input commentLimitString is null, an NumberFormatException would be thrown.